### PR TITLE
Feat adding template config

### PR DESCRIPTION
**Pull request by mliner**
# Description

Adding possibility to add template plugin config int coredns config

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Small minor change not affecting the Ansible Role code (Github Actions Workflow, Documentation etc.)

### CODE REVIEW

File: charts/node-local-dns/templates/configmap.yaml [hunk 1]
```diff
@@ -31,6 +31,31 @@ data:
         serve_stale
         {{- end }}
       }
+      {{- if $v.plugins.template }}
+      template {{ $v.plugins.template.class }} {{ $v.plugins.template.type }} {{- if $v.plugins.template.zone }} {{ $v.plugins.template.zone }} {{ end }} {
```
Comment (jaygridley) [inline charts/node-local-dns/templates/configmap.yaml:35]: Other plugins (cache, forward) are using `parameters` key to pass plugin arguments.

File: charts/node-local-dns/templates/configmap.yaml [hunk 2]
```diff
@@ -31,6 +31,31 @@ data:
...
+        {{- if $v.plugins.template.match }}
+        {{ $v.plugins.template.match }}
```
Comment (jaygridley) [inline charts/node-local-dns/templates/configmap.yaml:37]: Should the line be prefixed with `match` key?

File: charts/node-local-dns/templates/configmap.yaml [hunk 3]
```diff
@@ -31,6 +31,31 @@ data:
         serve_stale
         {{- end }}
       }
+      {{- if $v.plugins.template }}
```
Comment (jaygridley) [inline charts/node-local-dns/templates/configmap.yaml:34]: Add some example to the `values.yaml`

**Review by jaygridley — CHANGES_REQUESTED** [review]
I would like to have more context in the PR description why we are adding this.

File: charts/node-local-dns/values.yaml [hunk 4]
```diff
@@ -36,6 +36,14 @@ config:
         log:
           format: combined
           classes: all
+        template: {}
+          #  parameters: "ANY AAAA"
```
[the remark this answers is not in the document]
Comment (mliner) [reply charts/node-local-dns/values.yaml]: fixed

File: charts/node-local-dns/values.yaml [hunk 5]
```diff
@@ -36,6 +36,14 @@ config:
         log:
           format: combined
           classes: all
+        template: {}
```
[the remark this answers is not in the document]
Comment (mliner) [reply charts/node-local-dns/values.yaml]: added

File: charts/node-local-dns/templates/configmap.yaml (hunk 3 above)
Comment (mliner) [reply charts/node-local-dns/templates/configmap.yaml:34]: fixed

File: charts/node-local-dns/templates/configmap.yaml (hunk 1 above)
Comment (mliner) [reply charts/node-local-dns/templates/configmap.yaml]: changed

File: .github/workflows/pull-request.yaml
Comment (jaygridley) [inline .github/workflows/pull-request.yaml:1]: I would rename the file to `pr-release.yaml`

File: .github/workflows/pull-request.yaml [hunk 6]
```diff
@@ -0,0 +1,68 @@
+name: PR - release charts
```
Comment (jaygridley) [inline .github/workflows/pull-request.yaml:1]:
```suggestion
name: Pull Request - Release Charts
```

**Review by jaygridley — CHANGES_REQUESTED** [review]

**Review by jaygridley — APPROVED** [review]
LGTM

**Pull request merged**